### ISSUE TITLE
[0x80070002] Program error after upgrading itself

### ISSUE THREAD

**Issue by MMMMMoris** [t1m0thyj/WinDynamicDesktop#563]
- WDD Version: 5.4.1
- OS Version: Windows 11 Pro for Workstations 23H2 22631.3155

Steps to Reproduce:

1. Just run the game with PC startup
2. Look at the screen and wait for the error message.

Are you using the Microsoft Store version of WDD?: Yes/No No

Here is the error log
```
[2024-02-16 16:34:05.082] System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
   at WinDynamicDesktop.COM.IDesktopWallpaper.SetWallpaper(String monitorID, String wallpaper)
   at WinDynamicDesktop.WallpaperApi.SetWallpaper(String imagePath, Int32 displayIndex)
   at WinDynamicDesktop.UwpHelper.SetWallpaper(String imagePath, Int32 displayIndex)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()

[2024-02-17 03:24:59.016] System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
   at WinDynamicDesktop.COM.IDesktopWallpaper.SetWallpaper(String monitorID, String wallpaper)
   at WinDynamicDesktop.WallpaperApi.SetWallpaper(String imagePath, Int32 displayIndex)
   at WinDynamicDesktop.UwpHelper.SetWallpaper(String imagePath, Int32 displayIndex)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)

[2024-02-17 03:25:32.041] System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
   at WinDynamicDesktop.COM.IDesktopWallpaper.SetWallpaper(String monitorID, String wallpaper)
   at WinDynamicDesktop.WallpaperApi.SetWallpaper(String imagePath, Int32 displayIndex)
   at WinDynamicDesktop.UwpHelper.SetWallpaper(String imagePath, Int32 displayIndex)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
```

**Comment by t1m0thyj** [comment]
@MMMMMoris Thanks for reporting this. How many monitors/displays do you have and which theme do you have selected for each one?

**Comment by MMMMMoris** [comment]
> @MMMMMoris Thanks for reporting this. How many monitors/displays do you have and which theme do you have selected for each one?
> 
> Also if you navigate to [the folder where WinDynamicDesktop EXE is located](https://github.com/t1m0thyj/WinDynamicDesktop/wiki/Troubleshooting#finding-where-app-is-installed), there should be a "report.log" file that contains a more detailed error report. Please upload this file to the issue.

I have only on screen using. The theme I used is shuffle, and the one applied when I booted up was Mojave Desert.
The report.log is as below
[report.log](https://github.com/t1m0thyj/WinDynamicDesktop/files/14319008/report.log)

**Comment by MMMMMoris** [comment]
> * WDD Version: 5.4.1
> * OS Version: Windows 11 Pro for Workstations 23H2 22631.3155
> 
> Steps to Reproduce:
> 
> 1. Just run the game with PC startup
> 2. Look at the screen and wait for the error message.
> 
> Are you using the Microsoft Store version of WDD?: Yes/No No
> 
> Here is the error log
> 
> ```
> [2024-02-16 16:34:05.082] System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
>    at WinDynamicDesktop.COM.IDesktopWallpaper.SetWallpaper(String monitorID, String wallpaper)
>    at WinDynamicDesktop.WallpaperApi.SetWallpaper(String imagePath, Int32 displayIndex)
>    at WinDynamicDesktop.UwpHelper.SetWallpaper(String imagePath, Int32 displayIndex)
>    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
>    at System.Threading.ThreadPoolWorkQueue.Dispatch()
>    at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
> 
> [2024-02-17 03:24:59.016] System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
>    at WinDynamicDesktop.COM.IDesktopWallpaper.SetWallpaper(String monitorID, String wallpaper)
>    at WinDynamicDesktop.WallpaperApi.SetWallpaper(String imagePath, Int32 displayIndex)
>    at WinDynamicDesktop.UwpHelper.SetWallpaper(String imagePath, Int32 displayIndex)
>    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
>    at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
> 
> [2024-02-17 03:25:32.041] System.IO.FileNotFoundException: The system cannot find the file specified. (0x80070002)
>    at WinDynamicDesktop.COM.IDesktopWallpaper.SetWallpaper(String monitorID, String wallpaper)
>    at WinDynamicDesktop.WallpaperApi.SetWallpaper(String imagePath, Int32 displayIndex)
>    at WinDynamicDesktop.UwpHelper.SetWallpaper(String imagePath, Int32 displayIndex)
>    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
>    at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
> ```
> 
> ## This error just came out after the program upgraded itself yesterday.
> ## I cannot see what file specified the program is looking for.

### By the way, I have to mention that there was an error of my statement about the version of WDD. 
### The program I using currently is Microsoft Store version actually.

**Comment by t1m0thyj** [comment]
Thanks for providing details, will investigate this weekend. It looks like at the time of the error you had 2 active themes on multiple displays: `Windows_11` and `Win11Bloom23H2`. But the image files on disk seem to have been missing for the `Windows_11` theme so I suspect there may be a bug in WinDD that doesn't handle this "pre-installed" system theme properly.

**Comment by t1m0thyj** [comment]
Please disregard my earlier comment. After further investigation it seems like the error is caused by a bug in WinDD when theme shuffling is enabled. Please try the following steps to fix the issue:
* Open the Select Theme dialog and toggle the setting in "meatball menu" for shuffling themes to turn it off and back on.
* Check the selected theme for each display and if it is "Windows 11", change it to another theme that you have installed.
* Download the latest test build from [here](https://github.com/t1m0thyj/WinDynamicDesktop/actions/runs/7983958675). This should prevent WinDD from getting into this invalid state again.

**Comment by t1m0thyj** [comment]
Fixed in 5.4.2
